mpact/admin.py

```diff
@@ -39,3 +42,93 @@ class UserAdmin(BaseUserAdmin):
         ),
     )
     inlines = (UserProfileInline,)
+
+
+class CustomPeriodicTask(PeriodicTask):
+    message = models.TextField(
+        blank=True,
+        verbose_name="Message",
+        help_text="Enter the Message",
+    )
+
+
+class CustomPeriodicForm(PeriodicTaskForm):
+    individuals = [(indi.id, indi.first_name) for indi in Individual.objects.all()]
+    groups = [(chat.id, chat.title) for chat in Chat.objects.all()]
+
+    args = ChoiceField(
+        choices=[("chat", groups)] + [("individual", individuals)],
+        label="Chat",
+        required=False,
+    )
+
+    class Meta:
+        model = CustomPeriodicTask
+        exclude = ()
+
+    def clean_args(self):
+        val = self.cleaned_data["args"]
+        # Concatinating chat id and message and storing in args
+        self.cleaned_data["args"] = f"""[{val}, "{self.cleaned_data['message']}"]"""
+        return self._clean_json("args")
```

Comment:
What's the difference between `self._clean_json("args")` and the initial value of `self.cleaned_data["args"]`?

Reply:
self._clean_json is a function where we are passing the key(i.e., args). The initial value of self.cleaned_data["args"] is the chat_id from the drop-down and the later value contains the message as well.